bqskit/runtime/task.py

```diff
@@ -101,6 +101,11 @@ def start(self) -> None:
         """Initialize the task."""
         self.coro = self.run()
 
+    def cancel(self) -> None:
+        """Ask the coroutine to gracefully exit."""
+        if self.coro is not None:
+            self.coro.close()
```

Comment:
Is there ever an issue with a double-close? Should we throw an error here if the coro is None?

Reply:
Throwing an error if coro is `None` can be done simply by not checking if its `not None` and you will get something like "object of type NoneType does not have a method close()".

If you want to throw a more specific error we can do that.

When I had the check in `__del__` I had to not throw an error when `coro` is `None` because that seems to happen frequently.  I am new to the inner working of the bqskit runtime so I don't know what the conditions for `coro` being `None` are.

Reply:
Based on my read of the python documentation, I think closing a coroutine multiple times shouldn't be an issue, but I'll test it.

Reply:
Since this error is not something a user would be debugging we don't necessarily need to be more specific, but we can. A message like "Inactive RuntimeTask cannot be canceled." should be plenty for us to get some immediate insight.

I am curious, if you structure it as:

```python
if self.coro is None:
    raise RuntimeError("...")
self.coro.close()
```

If that error occurs in the usual cases, It shouldn't, right?